code/modules/surgery/dental_implant.dm

```diff
@@ -1,31 +1,49 @@
 /datum/surgery/dental_implant
 	name = "Dental implant"
-	steps = list(/datum/surgery_step/drill, /datum/surgery_step/insert_pill)
+	steps = list(/datum/surgery_step/drill, /datum/surgery_step/insert_object)
 	possible_locs = list(BODY_ZONE_PRECISE_MOUTH)
 
-/datum/surgery_step/insert_pill
-	name = "insert pill"
-	implements = list(/obj/item/reagent_containers/pill = 100)
+/datum/surgery_step/insert_object
+	name = "insert pill or signaling device"
+	implements = list(/obj/item/reagent_containers/pill = 100, /obj/item/assembly/signaler = 100)
 	time = 16
 
-/datum/surgery_step/insert_pill/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+/datum/surgery_step/insert_object/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	if(tool.type == /obj/item/assembly/signaler/)
+		var/obj/item/assembly/signaler/sig = tool
+		if(!sig.secured)
+			to_chat(user, "<span class='warning'>The [tool] is unsecured!</span>")
+			return -1
 	display_results(user, target, "<span class='notice'>You begin to wedge [tool] in [target]'s [parse_zone(target_zone)]...</span>",
-			"<span class='notice'>[user] begins to wedge \the [tool] in [target]'s [parse_zone(target_zone)].</span>",
+			"<span class='notice'>[user] begins to wedge the [tool] in [target]'s [parse_zone(target_zone)].</span>",
 			"<span class='notice'>[user] begins to wedge something in [target]'s [parse_zone(target_zone)].</span>")
 
-/datum/surgery_step/insert_pill/success(mob/user, mob/living/carbon/target, target_zone, obj/item/reagent_containers/pill/tool, datum/surgery/surgery, default_display_results = FALSE)
+/datum/surgery_step/insert_object/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
 	if(!istype(tool))
 		return 0
 
 	user.transferItemToLoc(tool, target, TRUE)
 
-	var/datum/action/item_action/hands_free/activate_pill/P = new(tool)
-	P.button.name = "Activate [tool.name]"
-	P.target = tool
-	P.Grant(target)	//The pill never actually goes in an inventory slot, so the owner doesn't inherit actions from it
+	if(istype(tool,/obj/item/reagent_containers/pill))
+		var/datum/action/item_action/hands_free/activate_pill/P = new(tool)
+		P.button.name = "Activate [tool.name]"
+		P.target = tool
+		P.Grant(target)	//The pill never actually goes in an inventory slot, so the owner doesn't inherit actions from it
+
+	else if(tool.type == /obj/item/assembly/signaler)
+		var/datum/action/item_action/hands_free/activate_signaler/S = new(tool)
+		S.button.name = "Activate [tool.name]"
+		S.target = tool
+		S.Grant(target)
+
+	else if(istype(tool,/obj/item/assembly/signaler/anomaly))
+		var/datum/action/item_action/hands_free/showoff/S = new(tool)
+		S.button.name = "Show off your [tool.name]"
+		S.target = tool
+		S.Grant(target)
```

Comment:
* These could stand to be redefined procs instead of bit istype check chains.
* Better variable names please.